server/api/repo.go

```diff
@@ -236,8 +237,21 @@ func PatchRepo(c *gin.Context) {
 	if in.AllowDeploy != nil {
 		repo.AllowDeploy = *in.AllowDeploy
 	}
-	if in.IsGated != nil {
-		repo.IsGated = *in.IsGated
+
+	if in.RequireApproval != nil {
+		if mode := model.ApprovalMode(*in.RequireApproval); mode.Valid() {
+			repo.RequireApproval = mode
+		} else {
+			c.String(http.StatusBadRequest, "Invalid require-approval setting")
+			return
+		}
+	} else if in.IsGated != nil { // TODO: remove isGated in next major release
+		log.Warn().Msgf("Deprecated 'gated' option was used, use 'require_approval' in the future to update an repo settings.")
```

Comment:
might have a better message or is this ok?

Reply:
The server admin does not need to be the repo admin, therefore this information would never reach the actual user and I would remove it.